sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -3781,6 +3781,14 @@ object SQLConf {
     .intConf
     .createWithDefault(0)
 
+  val PLANNED_WRITE_ENABLED = buildConf("spark.sql.plannedWrite.enabled")
```

Comment:
Indeed the name is not very descriptive. `Planned write` here means we want to explicitly plan file writes instead of adding various operations when executing the write. It could include things other than required ordering in the future. I am happy to brainstorm more here.